ios/RNEngageBySailthru.m

```diff
@@ -0,0 +1,273 @@
+
+#import "RNEngageBySailthru.h"
+#import <UserNotifications/UserNotifications.h>
+
+@interface MARPurchase ()
+
+- (nullable instancetype)initWithDictionary:(NSDictionary *)dictionary;
+
+@end
+
+@interface RNEngageBySailthru()
+
+@property (nonatomic, strong) EngageBySailthru *engageBySailthru;
+
+@end
+
+@implementation RNEngageBySailthru
+
+RCT_EXPORT_MODULE();
+
+- (instancetype)init {
+    self = [super init];
+    NSError *error;
+    _engageBySailthru = [self initializeEngageBySailthruWithError:&error];
```

Comment:
I recommend removing the `engageBySailthru` property and this initialisation in the `init` method. Instead we can call it every time a method needs the instance. You can pass in the `resolver` when one is available to allow the feedback to be returned to the JS side rather than just logging it out. You could also handle the error within the method, rather than passing it in.